src/Orchard/Environment/Extensions/Models/ExtensionDescriptor.cs

```diff
@@ -12,6 +12,8 @@ public class ExtensionDescriptor {
         /// </summary>
         public string Id { get; set; }
 
+        public string VirtualPath { get { return Location + "/" + Id; } }

```

Comment:
Actually there are several locations in the code where it is written 'VirtualPath = Location + "/" + Id'. and than VirtualPath is used.

Having explicit VirtualPath gives an explicit calculation instead of leaving the calculation to each implementation, it can reduced repetition and it can also be calculated once and used many times.